### PR TITLE
Navbar update

### PR DESCRIPTION
**Pull request by StreamSG**
Updated Navbar to offcanvas menu, added weather alert view and assorted code cleanup

### CODE REVIEW

File: src/app/services/weather.service.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { WeatherAlertResponse } from '../models/weather-alert.model';
   providedIn: 'root'
 })
 export class WeatherService {
-  private serverURL: string = 'https://api.weather.gov/alerts/active';
+  private serverURL: string = 'https://api.weather.gov/alerts';
```
Comment (looksliketyler) [inline src/app/services/weather.service.ts:12]: this needs set back to `/active` as this will break the current set up in master

File: src/app/components/views/weather-view/weather-view.component.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    <div class="row mb-5">
+      <div class="col text-center">
+        <div class="card pt-3">
+          <h3>Weather Alerts</h3>
+          <hr>
+          <h5>Active weather alerts in your area:</h5>
+        </div>
+      </div>
+    </div>
+    <ng-container *ngFor="let alert of weatherAlertResponse?.allActiveWeatherAlerts ; let i = index">
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.html:11]:
```suggestion
    <ng-container *ngFor="let alert of weatherAlertResponse?.allActiveWeatherAlerts; let i = index">
```

File: src/app/components/views/weather-view/weather-view.component.html [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    <div class="card mb-2" style="overflow:hidden;">
+        <div class="row">
+            <div class="col-3 text-center pt-3 rain" style="background-color:#444; color: white;">
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.html:14]: remove inline style

File: src/app/components/views/weather-view/weather-view.component.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<div class="container">
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.html:1]: this file has spacing of 4...please have tabs/indents be 2 spaces

File: src/app/components/views/weather-view/weather-view.component.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    <ng-container *ngFor="let alert of weatherAlertResponse?.allActiveWeatherAlerts ; let i = index">
+    <div class="card mb-2" style="overflow:hidden;">
+        <div class="row">
+            <div class="col-3 text-center pt-3 rain" style="background-color:#444; color: white;">
+                <p>ALERT</p>
+                <p>{{i+1}}</p>
+            </div>
+            <div class="col-9">
+                <h5>{{alert?.event}}</h5>
+                <p></p>
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.html:20]: what is this blank `<p>` here for?

File: src/app/components/views/weather-view/weather-view.component.scss [hunk 6]
```diff
@@ -0,0 +1,18 @@
+// .rain
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.scss:1]: this is all commented, it can be removed

File: src/app/components/views/weather-view/weather-view.component.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit } from '@angular/core';
+import { WeatherAlertResponse } from 'src/app/models/weather-alert.model';
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.ts:2]: local and package imports need separated

File: src/app/components/views/weather-view/weather-view.component.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import { WeatherService } from 'src/app/services/weather.service';
+
+@Component({
+  selector: 'app-weather-view',
+  templateUrl: './weather-view.component.html',
+  styleUrls: ['./weather-view.component.scss']
+})
+export class WeatherViewComponent implements OnInit {
+  public weatherAlertResponse: WeatherAlertResponse
+
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.ts:12]: remove extra line

File: src/app/components/views/weather-view/weather-view.component.ts [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+import { WeatherAlertResponse } from 'src/app/models/weather-alert.model';
+import { WeatherService } from 'src/app/services/weather.service';
+
+@Component({
+  selector: 'app-weather-view',
+  templateUrl: './weather-view.component.html',
+  styleUrls: ['./weather-view.component.scss']
+})
+export class WeatherViewComponent implements OnInit {
+  public weatherAlertResponse: WeatherAlertResponse
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.ts:11]: missing semi colon

File: src/app/components/views/weather-view/weather-view.component.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+
+  constructor(private weatherService: WeatherService) { }
+
+  ngOnInit(): void {
+
+    this.weatherService.getLoading().subscribe({
+      next: (loading) => {
+        if (!loading && this.weatherService.hasSuccessfullyCompleted()) {
+          // console.log(this.weatherService.getResults()); // TODO - remove console.log, for testing purposes
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.ts:21]: remove commented code

File: src/app/components/views/weather-view/weather-view.component.ts [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+})
+export class WeatherViewComponent implements OnInit {
+  public weatherAlertResponse: WeatherAlertResponse
+
+
+  constructor(private weatherService: WeatherService) { }
+
+  ngOnInit(): void {
+
+    this.weatherService.getLoading().subscribe({
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.ts:18]: please place this in its own method that is called in `ngOnInit()`.  it will need jsdocs and such

File: src/app/components/views/weather-view/weather-view.component.ts [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+      next: (loading) => {
+        if (!loading && this.weatherService.hasSuccessfullyCompleted()) {
+          // console.log(this.weatherService.getResults()); // TODO - remove console.log, for testing purposes
+          this.weatherAlertResponse = this.weatherService.getResults();
+        }
+      }
+    });
+
+}
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.ts:27]: spacing and indentation

**Comment by looksliketyler** [comment]
also, i cant comment on the photos, but was it decided to keep those as a banner?  remember, this will be seen by an AVP

File: src/app/services/weather.service.ts (hunk 1 above)
Comment (StreamSG) [reply src/app/services/weather.service.ts]: Fixed

File: src/app/components/views/weather-view/weather-view.component.html (hunk 2 above)
Comment (StreamSG) [reply src/app/components/views/weather-view/weather-view.component.html]: fixed!

File: src/app/components/views/weather-view/weather-view.component.html (hunk 4 above)
Comment (StreamSG) [reply src/app/components/views/weather-view/weather-view.component.html:1]: should be fixed now

File: src/app/components/views/weather-view/weather-view.component.html (hunk 5 above)
Comment (StreamSG) [reply src/app/components/views/weather-view/weather-view.component.html]: Good question. No answer.

File: src/app/components/views/weather-view/weather-view.component.scss (hunk 6 above)
Comment (StreamSG) [reply src/app/components/views/weather-view/weather-view.component.scss]: uncommented, it works but i was testing a few things.

File: src/app/components/views/weather-view/weather-view.component.ts (hunk 7 above)
Comment (StreamSG) [reply src/app/components/views/weather-view/weather-view.component.ts:2]: done

File: src/app/components/views/weather-view/weather-view.component.ts (hunk 11 above)
Comment (StreamSG) [reply src/app/components/views/weather-view/weather-view.component.ts:18]: working on it. Will mark resolved when i get home and can run and test to poke around and make sure it's done right

File: src/app/components/views/weather-view/weather-view.component.ts [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+  templateUrl: './weather-view.component.html',
+  styleUrls: ['./weather-view.component.scss']
+})
+export class WeatherViewComponent implements OnInit {
+  public weatherAlertResponse: WeatherAlertResponse;
+
+  constructor(private weatherService: WeatherService) { }
+
+  ngOnInit(): void {
+
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.ts:17]: remove blank spaces on line 17 and line 19

File: src/app/components/views/weather-view/weather-view.component.ts [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+    this.wsGetResults();
+
+}
+
+/**
+ * @description- looks like black magic to me
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.ts:23]: please update this jsdoc to state what the below method is doing

File: src/app/components/views/weather-view/weather-view.component.ts [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+ * @returns- something Micah coded, probably.
+ */
+
+  private wsGetResults(): any {
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.ts:27]: return statement for method says return type of `any`. what is this method returning?

also, please update the method name. `ws` doesnt tell me anything....you want a name to be descriptive as a jsdoc. `subscribeToWeatherService()`

File: src/app/components/views/weather-view/weather-view.component.scss [hunk 16]
```diff
@@ -0,0 +1,25 @@
+.card {
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.scss:1]: this file has four spaces instead of 2.

also, please add a blank space between each class. and i thought the .rain {} was being removed

File: src/app/components/views/weather-view/weather-view.component.html [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+        <h3>Weather Alerts</h3>
+        <hr>
+        <h5>Active weather alerts in your area:</h5>
+      </div>
+    </div>
+  </div>
+  <ng-container *ngFor="let alert of weatherAlertResponse?.allActiveWeatherAlerts; let i = index">
+    <div class="card mb-2">
+      <div class="row">
+        <div class="col-3 text-center pt-3 block rain">
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.html:14]: will @comm-scripter be submitting his own PR for the cards he was coding?  or are these staying. because i think we discussed having the cards look like the toast for consistency rather than the job cards

File: src/app/components/views/weather-view/weather-view.component.ts (hunk 15 above)
Comment (StreamSG) [reply src/app/components/views/weather-view/weather-view.component.ts:27]: put this in as a joke/placeholder while I was away

File: src/app/components/views/weather-view/weather-view.component.html (hunk 17 above)
Comment (StreamSG) [reply src/app/components/views/weather-view/weather-view.component.html:14]: I'm supposed to be getting with him tonight for this, we've discussed looking over and adding it, but I haven't gotten anything to add yet.

File: src/app/components/views/weather-view/weather-view.component.scss (hunk 16 above)
Comment (StreamSG) [reply src/app/components/views/weather-view/weather-view.component.scss:1]: I discussed having different animations for multiple weather alert types with Micah, we were going to see what it would take to be able to assign different animations to different weather alert types, but if you don't believe it's worth the time I'll just remove it.

File: src/app/app.component.scss [hunk 18]
```diff
@@ -0,0 +1 @@
+
```
[the remark this answers is not in the document]
Comment (StreamSG) [reply src/app/app.component.scss:1]: i have no idea

File: src/styles.scss [hunk 19]
```diff
@@ -3,6 +3,11 @@
 /* Importing Bootstrap SCSS file. */
 @import '~bootstrap/scss/bootstrap';
 
+.toast-container {
```
[the remark this answers is not in the document]
Comment (StreamSG) [reply src/styles.scss:6]: To get the Z index rules to properly apply, also felt redundant to have a position-absolute class when position: absolute can be applied to .toast-container

File: src/app/components/weather-alert/weather-alert.component.html [hunk 20]
```diff
@@ -1,4 +1,4 @@
-<div class="toast-container position-absolute">
+<div class="toast-container">
```
[the remark this answers is not in the document]
Comment (StreamSG) [reply src/app/components/weather-alert/weather-alert.component.html:1]: see styles.scss

File: src/app/components/header/header.component.scss [hunk 21]
```diff
@@ -1,7 +1,13 @@
...
\ No newline at end of file
+  position: absolute;
+  left: 50%;
+  margin-left: -35px !important;  /* 50% of your logo width */
+  top: 10px;
+  display: block;
+}
+
+.offcanvas {
+
```
Comment (looksliketyler) [inline src/app/components/header/header.component.scss:10]:
```suggestion
```

File: src/app/components/views/weather-view/weather-view.component.html [hunk 22]
```diff
@@ -0,0 +1,24 @@
...
+    <div class="col text-center">
+      <div class="card pt-3">
+        <h3>Weather Alerts</h3>
+        <hr>
+        <h5>Active weather alerts in your area:</h5>
+      </div>
+    </div>
+  </div>
+</div>
+<ng-container *ngFor="let alert of weatherAlertResponse?.allActiveWeatherAlerts ; let i = index">
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.html:12]:
```suggestion
<ng-container *ngFor="let alert of weatherAlertResponse?.allActiveWeatherAlerts; let i = index">
```

File: src/app/components/views/weather-view/weather-view.component.ts [hunk 23]
```diff
@@ -0,0 +1,34 @@
...
+  public weatherAlertResponse: WeatherAlertResponse;
+
+  constructor(private weatherService: WeatherService) { }
+
+  ngOnInit(): void {
+    this.subscribeToWeatherService();
+}
+
+/**
+ * @description Subscribes to weather service and returns array for populating view
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.ts:21]:
```suggestion
 * @description - Subscribes to weather service and sets global variable for the api response
```

File: src/app/components/views/weather-view/weather-view.component.ts [hunk 24]
```diff
@@ -0,0 +1,34 @@
...
+
+  ngOnInit(): void {
+    this.subscribeToWeatherService();
+}
+
+/**
+ * @description - Subscribes to weather service and sets global variable for the api response
+ * @returns {void}
+ */
+
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.ts:24]:
```suggestion
```

File: src/app/components/views/weather-view/weather-view.component.ts [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+  ngOnInit(): void {
+    this.subscribeToWeatherService();
+}
+
+/**
+ * @description - Subscribes to weather service and sets global variable for the api response
+ * @returns {void}
+ */
+  private subscribeToWeatherService(): any {
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.ts:24]:
```suggestion
  private subscribeToWeatherService(): void {
```

File: src/app/components/views/weather-view/weather-view.component.ts [hunk 26]
```diff
@@ -0,0 +1,33 @@
...
+    this.subscribeToWeatherService();
+}
+
+/**
+ * @description - Subscribes to weather service and sets global variable for the api response
+ * @returns {void}
+ */
+  private subscribeToWeatherService(): void {
+    this.weatherService.getLoading().subscribe({
+       next: (loading) => {
```
Comment (looksliketyler) [inline src/app/components/views/weather-view/weather-view.component.ts:26]:
```suggestion
       next: (loading: boolean) => {
```

File: src/app/services/weather.service.ts [hunk 27]
```diff
@@ -100,6 +100,25 @@ export class WeatherService {
             this.updateLoading(false);
           }
         });
+    if (!this.loading && lat && long) {
+      this.updateLoading(true);
+      this.httpSubscription = this.http.get(`${this.serverURL}?point=44.0805%2C-103.2310&limit=500`)//got rid of ${lat} and long
```
Comment (looksliketyler) [inline src/app/services/weather.service.ts:105]: this needs fixed - lat/long is hardcoded

File: src/app/services/weather.service.ts [hunk 28]
```diff
@@ -100,6 +100,25 @@ export class WeatherService {
...
+        .subscribe({
+          next: (apiResponse: any) => {
+            console.log(apiResponse); // TODO - remove, for testing purposes
+            this.apiResults = new WeatherAlertResponse(apiResponse);
+            console.log('api as new class -', this.apiResults);
```
Comment (looksliketyler) [inline src/app/services/weather.service.ts:110]: remove console.log

File: src/app/services/weather.service.ts [hunk 29]
```diff
@@ -100,6 +100,25 @@ export class WeatherService {
             this.updateLoading(false);
           }
         });
+    if (!this.loading && lat && long) {
+      this.updateLoading(true);
+      this.httpSubscription = this.http.get(`${this.serverURL}?point=44.0805%2C-103.2310&limit=500`)//got rid of ${lat} and long
+        .subscribe({
+          next: (apiResponse: any) => {
+            console.log(apiResponse); // TODO - remove, for testing purposes
```
Comment (looksliketyler) [inline src/app/services/weather.service.ts:108]: remove console.log

File: src/app/services/weather.service.ts [hunk 30]
```diff
@@ -100,6 +100,25 @@ export class WeatherService {
...
+            this.apiResults = new WeatherAlertResponse(apiResponse);
+            console.log('api as new class -', this.apiResults);
+            this.isSuccessfullyCompleted = true;
+            this.updateLoading(false);
+          },
+          error: (error: any) => {
+            this.apiResults = new WeatherAlertResponse(error.error);
+            console.error(error);
```
Comment (looksliketyler) [inline src/app/services/weather.service.ts:116]: remove console.log

**Review by looksliketyler — CHANGES_REQUESTED** [review]
i changed a few minor things, but please see comments that still need addressed

File: src/app/services/weather.service.ts [hunk 31]
```diff
@@ -100,6 +100,25 @@ export class WeatherService {
             this.updateLoading(false);
           }
         });
+    if (!this.loading && lat && long) {
```
Comment (looksliketyler) [inline src/app/services/weather.service.ts:103]: @StreamSG in testing stuff - i found the logic is duplicated, probably from a merge conflict resolution...simple deletion needed pal

File: src/app/services/weather.service.ts (hunk 27 above)
Comment (StreamSG) [reply src/app/services/weather.service.ts:105]: damn it kerry

**Review by looksliketyler — APPROVED** [review]
